### PR TITLE
util: make ScheduleBatchPriority advisory only

### PR DESCRIPTION
**Pull request by fanquake**

### CODE REVIEW

File: src/util/system.h [hunk 1]
```diff
@@ -367,8 +367,8 @@ std::string CopyrightHolders(const std::string& strPrefix);
  * On platforms that support it, tell the kernel the calling thread is
  * CPU-intensive and non-interactive. See SCHED_BATCH in sched(7) for details.
  *
- * @return The return value of sched_setschedule(), or 1 on systems without
- * sched_setschedule().
+ * @return The return value of pthread_setschedparam(), or 1 on systems without
+ * SCHED_BATCH.
```
Comment (laanwj) [inline src/util/system.h]: I don't think it's necessary to describe implementation details here at all. Maybe just state it returns 0 on success and any other value on failure or when the functionality is not supported on the OS.

Note that we don't use the return value at all, not even for logging.

I think I'd prefer making this a purely advisory function, that returns `void`.

Comment (fanquake) [reply src/util/system.h]: > I think I'd prefer making this a purely advisory function, that returns void.

I've updated the PR to do that.

**Comment by laanwj** [comment]
ACK d2a3a5cadbe58c0fe363bbc6acac293d41eedf7e

**Comment by practicalswift** [comment]
@fanquake The description for this PR is empty: can you describe why this change is needed? :)

**Comment by laanwj** [comment]
It removes an unused return value, where a weird return convention was used. Instead of documenting it, it was better to remove the return value completely.

**Pull request merged**